src/sagemaker_tensorflow_container/training.py

```diff
@@ -13,16 +13,127 @@
 
 from __future__ import absolute_import
 
+import json
 import logging
 
 import sagemaker_containers.beta.framework as framework
 
+
 logger = logging.getLogger(__name__)
 
 
+SAGEMAKER_PARAMETER_SERVER_NUM = 'sagemaker_parameter_server_num'
+
+
+def _is_host_master(hosts, current_host):
+    return current_host == hosts[0]
+
+
+def _build_tf_config(hosts, current_host, ps_num=0, ps_task=False):
+    """
+    Builds a dictionary containing cluster information based on number of hosts and number of
+    parameter servers.
+
+    Args:
+        hosts (list[str]): List of host name in the cluster
+        current_host (str): Current host name
+        ps_num (int): Number of parameter servers
+        ps_task (bool): Set to True if this config is built for a ps server process
+
+    Returns:
+        dict[str: dict]: A dictionary describing the cluster setup for distributed training.
+        For more information regarding TF_CONFIG:
+        https://cloud.google.com/ml-engine/docs/tensorflow/distributed-training-details
+
+    """
+
+    # Assign the first host as the master. Rest of the hosts if any will be worker hosts.
+    # The first ps_num hosts will also have a parameter task assign to them.
+    masters = hosts[:1]
+    workers = hosts[1:]
+    ps = hosts[:ps_num] if len(hosts) > 1 and ps_num > 0 else None
+
+    def host_addresses(hosts, port=2222):
+        return ["{}:{}".format(host, port) for host in hosts]
+    tf_config = {
+        "cluster": {
+            "master": host_addresses(masters)
+        },
+        "environment": "cloud"
+    }
+
+    if ps is not None:
+        tf_config["cluster"]["ps"] = host_addresses(ps, port='2223')
+
+    if len(workers) > 0:
+        tf_config["cluster"]["worker"] = host_addresses(workers)
+
+    if ps_task:
+        if ps is None:
+            raise ValueError(
+                "Can not have a ps task if there are no parameter servers in the cluster")
+        task_type = "ps"
+        task_index = ps.index(current_host)
+    elif _is_host_master(hosts, current_host):
+        task_type = "master"
+        task_index = 0
+    else:
+        task_type = "worker"
+        task_index = workers.index(current_host)
+
+    tf_config["task"] = {"index": task_index, "type": task_type}
+    return tf_config
+
+
+def _get_env_vars_with_tf_config(env, ps_task):
```

Comment:
I think prefixing with "get" is considered a little unpythonic or whatever - anyway, you can make it just `_env_vars_with_tf_config`